src/Covenant/Constant.hs

```diff
@@ -107,6 +108,27 @@ data AConstant
       Show
     )
 
+-- | The type of Plutus constant terms.
+--
+-- @since 1.0.0
+data TyConstant
```

Comment:
This is slightly misnamed: `ByteString`s for example aren't _necessarily_ constants, as we can construct them from other operations (such as `ConsByteString`). From what I can see later-on, this seems to be a general 'type of Covenant expression', which means it's missing a couple of things (such as BLS types for example).